examples/object_detection/ios/ObjectDetector/ViewContollers/CameraViewController.swift

```diff
@@ -0,0 +1,256 @@
+  private lazy var cameraCapture = CameraFeedManager(previewView: previewView)
+  
+  private let objectDetectorServiceQueue = DispatchQueue(
+    label: "objectDetectorServiceQueue",
+    attributes: .concurrent)
+  
+  // Queuing reads and writes to objectDetectorService using the Apple recommended way
+  // as they can be read and written from multiple threads and can result in race conditions.
+  private var _objectDetectorService: ObjectDetectorService?
+  private var objectDetectorService: ObjectDetectorService? {
+    get {
+      objectDetectorServiceQueue.sync {
+        return self._objectDetectorService
+      }
+    }
+    set {
+      objectDetectorServiceQueue.async(flags: .barrier) {
+        self._objectDetectorService = newValue
+      }
+    }
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    print("CameraController Appear")
+#if !targetEnvironment(simulator)
+    initializeObjectDetectorServiceOnSessionResumption()
+    cameraCapture.checkCameraConfigurationAndStartSession()
+#endif
+  }
+  
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    print("CameraController Disappear")
+#if !targetEnvironment(simulator)
+    cameraCapture.stopSession()
+    clearObjectDetectorServiceOnSessionInterruption()
+#endif
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+#if !targetEnvironment(simulator)
+    cameraCapture.delegate = self
+#endif
+    // Do any additional setup after loading the view.
+  }
+  
+  @IBAction func onClickResume(_ sender: Any) {
+    if isSessionRunning {
+      self.resumeButton.isHidden = true
+      self.cameraUnavailableLabel.isHidden = true
+    }
+  }
+  
+  // Resume camera session when click button resume
+  @IBAction func resumeButtonTouchUpInside(_ sender: Any) {
+    cameraCapture.resumeInterruptedSession {[weak self] isSessionRunning in
+      if isSessionRunning {
+        self?.resumeButton.isHidden = true
+        self?.cameraUnavailableLabel.isHidden = true
+        self?.initializeObjectDetectorServiceOnSessionResumption()
+      }
+    }
+  }
+  
+  private func initializeObjectDetectorServiceOnSessionResumption() {
+    clearAndInitializeObjectDetectorService()
+    addDetectorMetadataKeyValueObservers()
+  }
+  
+  private func clearAndInitializeObjectDetectorService() {
+    objectDetectorService = nil
+      objectDetectorService = ObjectDetectorService
+        .liveStreamDetectorService(
+          model: DetectorMetadata.sharedInstance.model,
+          maxResults: DetectorMetadata.sharedInstance.maxResults,
+          scoreThreshold: DetectorMetadata.sharedInstance.scoreThreshold,
+          liveStreamDelegate: self)
+  }
+  
+  private func clearObjectDetectorServiceOnSessionInterruption() {
+    removeDetectorMetadataKeyValueObservers()
+    objectDetectorService = nil
+  }
+}
+
+extension CameraViewController: CameraFeedManagerDelegate {
+  override func observeValue(
```

Comment:
Move this method because it doesn't belong to the CameraFeedManagerDelegate protocol.